api/src/services/users/delete_saved_search.py

```diff
@@ -1,20 +1,20 @@
 from uuid import UUID
 
-from sqlalchemy import select
+from sqlalchemy import update
 
 from src.adapters import db
 from src.api.route_utils import raise_flask_error
 from src.db.models.user_models import UserSavedSearch
 
 
 def delete_saved_search(db_session: db.Session, user_id: UUID, saved_search_id: UUID) -> None:
-    saved_search = db_session.execute(
-        select(UserSavedSearch).where(
+    result = db_session.execute(
+        update(UserSavedSearch)
+        .where(
             UserSavedSearch.saved_search_id == saved_search_id, UserSavedSearch.user_id == user_id
         )
-    ).scalar_one_or_none()
+        .values(is_deleted=True)
```

Comment:
Could we leave this as it was before? Just instead of deleting it, we set the value on the saved search record returned?